Project/core/src/main/java/View/screen/GameScreenComponents/TurnBar.java

```diff
@@ -0,0 +1,127 @@
+package View.screen.GameScreenComponents;
+
+import java.util.Set;
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.scenes.scene2d.ui.Label;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.utils.Align;
+import java.util.HashSet;
+import Controller.GameController;
+
+/**
+ * The TurnBar class represents a UI component that displays the current turn and tracks
+ * previous turns in a game.
+ * This class is responsible for updating the turn display and managing revealed turns.
+ */
+
+public class TurnBar extends Table {
+    private final GameController gameController;
+    private String timeValue; // The current time value as a string
+    private final Label timeTracker; // Label for displaying the current time
+    private final Table pastTurn; // Table for displaying past turns
+    private final Set<Table> revealedTurns; // Set of tables representing revealed turns
+    private final Set<Integer> revealedTurnsInt; // Set of integers representing revealed turn indices
+    private final Skin skin;
+
+    /**
+     * Constructor for TurnBar.
+     * Initializes the UI elements and prepares the turn bar for updates.
+     *
+     * @param gameController The GameController that manages game state.
+     */
+
+    public TurnBar(GameController gameController) {
+        this.setDebug(true); // Enables debug mode for visualizing layout borders
+        this.gameController = gameController;
+        skin = new Skin(Gdx.files.internal("comicui/comic-ui.json"));
```

Comment:
please use the asset manager! :D

it exists in the application, "application.assets"

Reply:
I already added it.